plugins/vpc-bridge/config/k8s/pipe.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	winio "github.com/Microsoft/go-winio"
+	log "github.com/cihub/seelog"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	//PipeConnectionTimeout Represents timeout for reading from connection object.
+	PipeConnectionTimeout = 1 * time.Minute
+
+	//NamedPipePathPrefix Represents prefix for named pipe path.
+	NamedPipePathPrefix = `\\.\pipe\aws-vpc-k8s-connector-%s`
+)
+
+// PipeReadResult Represents output received from named pipe.
+type PipeReadResult struct {
+	stdout string // Represents message with output data.
+	stderr error  // Represents error while getting output.
+}
+
+// PipeOperator Represents interface with all operations on named piped.
+type PipeOperator interface {
```

Comment:
Why the interface and the struct? Do we need both?

Reply:
Keeping both interface and struct as struct has additional fields and funcs which don't need to be used directly outside of `pipe.go`. We are refering pipe in other files via `pipeOperator ` interface.
I have changed these structs and interface to unexported.